### PR TITLE
Chainbridgenetwork

### PR DESCRIPTION
**Pull request by ChainBridgeNetworkTeam**
# Grant Application Checklist

- [x] The [application-template.md](https://github.com/w3f/Open-Grants-Program/blob/master/applications/application-template.md) has been copied, renamed ( "project_name.md") and updated.
- [x] A BTC or Ethereum (DAI) address for the payment of the milestones is provided inside the application.  
- [x] The software of the project will be released under the Apache license version 2.0 as specified in the [terms and conditions](https://gist.github.com/Noc2/75bc58e8ce9b5d419ff883b0cf2b8c19).
- [x] The total funding amount of the project is below USD $30k for initial grant applications and $100k for follow-up grants.
- [x] The initial PR contains only one commit (squash if needed before submitting your PR).
- [x] The grant will only be announced once we've successfully [delivered](https://github.com/w3f/Grant-Milestone-Delivery) the first milestone.

### CODE REVIEW

**Review by Noc2 — CHANGES_REQUESTED** [review]
Thanks for the application. But your application is currently above the 30k limit for initial grant applications. Also it looks like your first milestone currently only contains documentation and a license. Could you please update the deliveries to be more specific? Also how is your solution different from the polkadot.js extension https://github.com/polkadot-js/extension, since enzyme is no longer actively maintained at the moment?  How do you plan to maintain the wallet?

**Comment by ChainBridgeNetworkTeam** [comment]
> Thanks for the application. But your application is currently above the 30k limit for initial grant applications. Also it looks like your first milestone currently only contains documentation and a license. Could you please update the deliveries to be more specific? Also how is your solution different from the polkadot.js extension https://github.com/polkadot-js/extension, since enzyme is no longer actively maintained at the moment? How do you plan to maintain the wallet?

Thank you very much for your feedback. I am sorry to reply to you so late. We have updated the application documents for several issues you mentioned.

1) The initial grant exceeds the limit: We have updated the grant amount in the "Development Roadmap-Overview" of the application document so that the grant meets the limit of no more than 30,000 US dollars

2) Milestone delivery time: We have updated the description of delivery time and how to verify it in the "Development Roadmap-Milestone-M1/M2/M3" of the application document（In particular, we have completed the first milestone and the extension has been listed on the Google expansion market. You can [install Doter](https://chrome.google.com/webstore/detail/doter/abamjefkidngfegdjbmffdmbgjgpaobf) right away and verify the first milestone.）

3) The difference between our solution and the polkadot.js extension:
Polkadot.js extension is an official account management tool, Compared with polkadot.js extension, Doter's positioning is not only an account management tool. Doter will also implements a series of common functions in Polkadot ecology, such as Transfer and receive, query transaction records, wallet management, easy-to-operate on-chain governance modules, staking modules, etc. This means that doter can not only interact well with other dapps, but also independently complete the functions mentioned above, providing users with a one-stop experience, which polkadot.js extension does not have.   <br/>
In addition, Doter will also provide a completely user-centric UI experience to make it easier for users to participate in the Polkadot ecosystem.

4) How to maintain the wallet: 
In the near future, Our team will have someone in charge of maintaining the Doter, Update at least once a month to fix bugs or improve the experience，to making Doter as close as possible to mature browser extension in other ecosystems（such as Ethereum）

Simultaneously, we will build our own community as soon as possible. We believe that a solid community is the foundation for Doter to seize the market, Through word of mouth from seed users, we will accumulate the first batch of users. After completing all the milestones, we will place advertisements in a number of blockchain media, and rapidly expand the number of users through advertising. Secondly, we will embed a frequently used DApp in the wallet and cultivate user habits through this, so that Doter will gradually become the portal of the Polkadot ecosystem.

In summary, we have appropriate market timing, clear product direction, reliable development progress, friendly UI experience, detailed operation planning, and the most important thing is that we have a reliable team, We have 3 Years of work experience in NASDAQ listed company, we have the ability to make Doter a competitive product.

Finally, thank you again for your feedback and look forward to your other valuable comments!

**Comment by ChainBridgeNetworkTeam** [comment]
> Thanks for the application. But your application is currently above the 30k limit for initial grant applications. Also it looks like your first milestone currently only contains documentation and a license. Could you please update the deliveries to be more specific? Also how is your solution different from the polkadot.js extension https://github.com/polkadot-js/extension, since enzyme is no longer actively maintained at the moment? How do you plan to maintain the wallet?

Thank you very much for your feedback. I am sorry to reply to you so late. We have updated the application documents for several issues you mentioned.

1) The initial grant exceeds the limit: We have updated the grant amount in the "Development Roadmap-Overview" of the application document so that the grant meets the limit of no more than 30,000 US dollars

2) Milestone delivery time: We have updated the description of delivery time and how to verify it in the "Development Roadmap-Milestone-M1/M2/M3" of the application document（In particular, we have completed the first milestone and the extension has been listed on the Google expansion market. You can [install Doter](https://chrome.google.com/webstore/detail/doter/abamjefkidngfegdjbmffdmbgjgpaobf) right away and verify the first milestone.）

3) The difference between our solution and the polkadot.js extension:
Polkadot.js extension is an official account management tool, Compared with polkadot.js extension, Doter's positioning is not only an account management tool. Doter will also implements a series of common functions in Polkadot ecology, such as Transfer and receive, query transaction records, wallet management, easy-to-operate on-chain governance modules, staking modules, etc. This means that doter can not only interact well with other dapps, but also independently complete the functions mentioned above, providing users with a one-stop experience, which polkadot.js extension does not have.   <br/>
In addition, Doter will also provide a completely user-centric UI experience to make it easier for users to participate in the Polkadot ecosystem.

4) How to maintain the wallet: 
In the near future, Our team will have someone in charge of maintaining the Doter, Update at least once a month to fix bugs or improve the experience，to making Doter as close as possible to mature browser extension in other ecosystems（such as Ethereum）

Simultaneously, we will build our own community as soon as possible. We believe that a solid community is the foundation for Doter to seize the market, Through word of mouth from seed users, we will accumulate the first batch of users. After completing all the milestones, we will place advertisements in a number of blockchain media, and rapidly expand the number of users through advertising. Secondly, we will embed a frequently used DApp in the wallet and cultivate user habits through this, so that Doter will gradually become the portal of the Polkadot ecosystem.

In summary, we have appropriate market timing, clear product direction, reliable development progress, friendly UI experience, detailed operation planning, and the most important thing is that we have a reliable team, We have 3 Years of work experience in NASDAQ listed company, we have the ability to make Doter a competitive product.

Finally, thank you again for your feedback and look forward to your other valuable comments!

**Comment by ChainBridgeNetworkTeam** [comment]
> Thanks for the application. But your application is currently above the 30k limit for initial grant applications. Also it looks like your first milestone currently only contains documentation and a license. Could you please update the deliveries to be more specific? Also how is your solution different from the polkadot.js extension https://github.com/polkadot-js/extension, since enzyme is no longer actively maintained at the moment? How do you plan to maintain the wallet?

Thank you very much for your feedback. I am sorry to reply to you so late. We have updated the application documents for several issues you mentioned.

1) The initial grant exceeds the limit: We have updated the grant amount in the "Development Roadmap-Overview" of the application document so that the grant meets the limit of no more than 30,000 US dollars

2) Milestone delivery time: We have updated the description of delivery time and how to verify it in the "Development Roadmap-Milestone-M1/M2/M3" of the application document（In particular, we have completed the first milestone and the extension has been listed on the Google expansion market. You can [install Doter](https://chrome.google.com/webstore/detail/doter/abamjefkidngfegdjbmffdmbgjgpaobf) right away and verify the first milestone.）

3) The difference between our solution and the polkadot.js extension:
Polkadot.js extension is an official account management tool, Compared with polkadot.js extension, Doter's positioning is not only an account management tool. Doter will also implements a series of common functions in Polkadot ecology, such as Transfer and receive, query transaction records, wallet management, easy-to-operate on-chain governance modules, staking modules, etc. This means that doter can not only interact well with other dapps, but also independently complete the functions mentioned above, providing users with a one-stop experience, which polkadot.js extension does not have.   <br/>
In addition, Doter will also provide a completely user-centric UI experience to make it easier for users to participate in the Polkadot ecosystem.

4) How to maintain the wallet: 
In the near future, Our team will have someone in charge of maintaining the Doter, Update at least once a month to fix bugs or improve the experience，to making Doter as close as possible to mature browser extension in other ecosystems（such as Ethereum）

Simultaneously, we will build our own community as soon as possible. We believe that a solid community is the foundation for Doter to seize the market, Through word of mouth from seed users, we will accumulate the first batch of users. After completing all the milestones, we will place advertisements in a number of blockchain media, and rapidly expand the number of users through advertising. Secondly, we will embed a frequently used DApp in the wallet and cultivate user habits through this, so that Doter will gradually become the portal of the Polkadot ecosystem.

In summary, we have appropriate market timing, clear product direction, reliable development progress, friendly UI experience, detailed operation planning, and the most important thing is that we have a reliable team, We have 3 Years of work experience in NASDAQ listed company, we have the ability to make Doter a competitive product.

Finally, thank you again for your feedback and look forward to your other valuable comments!

**Review by Noc2 — CHANGES_REQUESTED** [review]
Thanks for the reply and updating the application. I have a few follow-up questions: 

- It’s definitely nice to see that you are already working on the project and that you completed milestone 1 already. This increases your chances of getting a grant. But I suggest removing it from the application itself. Instead you could integrate additional deliverables or an additional milestone at the end if you want to. 
- Could you integrate tests into the application? 
- How do you ensure the security of your wallet? 
- Do you also plan to support kusama? Which additional networks are you planning to support?

**Comment by ChainBridgeNetworkTeam** [comment]
@Noc2  Hello, thank you again for your previous feedback! We are waiting for your further reply in order to adjust our application if necessary. Your previous questions have been answered in the last reply.

**Review by Noc2 — CHANGES_REQUESTED** [review]
Sorry for the late reply. Could you add unit tests to the deliverables. Also could you try to make the Deliverables and Specification more concrete. For example, for milestone 3, currently only the headline says “ kusama network” and it’s mentioned in the specification or could you link to background.js, etc.. Also is it correct that you won’t support staking governance etc. for kusama?

**Comment by ChainBridgeNetworkTeam** [comment]
> Sorry for the late reply. Could you add unit tests to the deliverables. Also could you try to make the Deliverables and Specification more concrete. For example, for milestone 3, currently only the headline says “ kusama network” and it’s mentioned in the specification or could you link to background.js, etc.. Also is it correct that you won’t support staking governance etc. for kusama?  

@Noc2 , hello! In order to make our application clear and easy to verify, we adjusted our milestones. In this version of the application, we will complete account creation, transfer, transaction record query, account injection, signature, democratic governance (proposal voting) and other functions in milestone 1.  In milestone 2, we will support these functions in the Kusama network. Staking, Committee election and other functions will be implemented in subsequent updates, but we will not include them in this application.

**Comment by Noc2** [comment]
Thanks for the update and quick reply. Could you list all the functions, which you plan to support for both milestones under the Core component deliverable? It’s currently not so clear, for example only milestone 1 mentions governance.

**Comment by ChainBridgeNetworkTeam** [comment]
> Thanks for the update and quick reply. Could you list all the functions, which you plan to support for both milestones under the Core component deliverable? It’s currently not so clear, for example only milestone 1 mentions governance.  

@Noc2 Thank you for your reply! We have adjusted the details of milestone in the application, and the contents to be delivered have been clearly listed too.  

In milestone 1, the deliveries are:  
* Wallet injection on Polkadot
* Transaction signature on Polkadot  

In milestoe 2, the deliveries are:  
* Create and import wallets on Kusama
* Transfer and receive on Kusama
* Backup wallet on Kusama
* Query transaction records on Kusama
* Referendum on the chain on Kusama
* Switch multiple wallets on Kusama
* Manage address book on Kusama
* Switch language on Kusama
* wallet injection and signature on Kusama

**Comment by Noc2** [comment]
Thanks again for the update. What do you mean with “Transaction signature on Polkadot”? Do you support signing of all transactions directly in the extension (governance, multisig, etc)?

**Comment by ChainBridgeNetworkTeam** [comment]
> Thanks again for the update. What do you mean with “Transaction signature on Polkadot”? Do you support signing of all transactions directly in the extension (governance, multisig, etc)?

@Noc2 Yes, Doter will provide an experience similar to Polkadot. {js}, which is to call up a pop-up window for users to sign when the transfer operation occurs

**Comment by Noc2** [comment]
So, do you still need a separate UI like polkadot.js/app or does your app support this inside your extension?

**Comment by ChainBridgeNetworkTeam** [comment]
> polkadot.js/app

@Noc2 , Hello! Doter itself supports signature popups. With the help of package @polkadot/extension-inject, Doter can interact with any DAPP that use package @polkadot/extension-dapp

**Comment by Noc2** [comment]
Okay, so you can not vote or stake with only your app installed after milestone 1, correct? The main reason I’m asking is that only supporting the “injection on Polkadot” seems to me a little bit expensive for milestone 1. So, I suggest integrating other deliverables, which your application already supports anyway. Also the costs of milestone 2 seem to be wrong (Costs: $95,00).

**Comment by ChainBridgeNetworkTeam** [comment]
> Okay, so you can not vote or stake with only your app installed after milestone 1, correct? The main reason I’m asking is that only supporting the “injection on Polkadot” seems to me a little bit expensive for milestone 1. So, I suggest integrating other deliverables, which your application already supports anyway. Also the costs of milestone 2 seem to be wrong (Costs: $95,00).

> Okay, so you can not vote or stake with only your app installed after milestone 1, correct? The main reason I’m asking is that only supporting the “injection on Polkadot” seems to me a little bit expensive for milestone 1. So, I suggest integrating other deliverables, which your application already supports anyway. Also the costs of milestone 2 seem to be wrong (Costs: $95,00).

@Noc2 In the current two milestones, Doter don’t implement vote and staking in Polkadot ecosystem, so we adjusted the grant amount we applied for. Besides, in the first milestone, we will optimize the experience of Doter, which is already online in the google extension store. Relevant information has been added in the application, too.  

We feel so sorry for wrong amount of milestone 2, and has modified it. The correct amount should be $9,500

**Comment by Noc2** [comment]
Thanks for the update. I will share the application with the rest of the team. Could you provide a link to your twitter account, I couldn’t find it. Apart from this, maybe we can quickly jump on a call. Feel free to send email to grants@web3.foundation to organize a meeting. Usually that isn’t mandatory, but given that you are developing a wallet it would be nice to quickly connect.

**Comment by BenWhiteJam** [comment]
@ChainBridgeNetworkTeam thanks for the application!
You mention the launch of Polkadot Parachains in your intro, but I couldn't really find features that would support parachain launches nor cross-consensus message passing in specifics. Maybe you can elaborate on if these features would be part, how these logic components could look like? etc.?

**Comment by ChainBridgeNetworkTeam** [comment]
> Thanks for the update. I will share the application with the rest of the team. Could you provide a link to your twitter account, I couldn’t find it. Apart from this, maybe we can quickly jump on a call. Feel free to send email to [grants@web3.foundation](mailto:grants@web3.foundation) to organize a meeting. Usually that isn’t mandatory, but given that you are developing a wallet it would be nice to quickly connect.

Thank you! Our twitter link is [https://twitter.com/chainBridgeNet1](https://twitter.com/chainBridgeNet1), and my email is [dianluyuanli@foxmail.com](dianluyuanli@foxmail.com), if there are any other questions, you can concat me with this email，I will reply as soon as possible. I'll send an email later to discuss the meeting.

**Comment by ChainBridgeNetworkTeam** [comment]
> @ChainBridgeNetworkTeam thanks for the application!
> You mention the launch of Polkadot Parachains in your intro, but I couldn't really find features that would support parachain launches nor cross-consensus message passing in specifics. Maybe you can elaborate on if these features would be part, how these logic components could look like? etc.?

We are sorry, maybe our expression made you confused. We just mentioned the development of parachain in Polkadot ecosystem in the application. Our Chrome extension itself does not involve any parachain feature. Sorry again.

**Review by Noc2 — APPROVED** [review]

**Review by gautamdhameja — APPROVED** [review]

**Review by BenWhiteJam — APPROVED** [review]
